spec/e2e/has_one_spec.rb

```diff
@@ -206,6 +208,41 @@
       manager.subordinates << employee
       expect(employee.manager(rel_length: 1)).to eq(manager)
     end
+
+    # since chainable: true is an option, this test both checks to see that default options
+    # are honored, and checks to make sure the provided options are merged into the default options
+    it 'honors default options' do
+      expect(
+        employee.manager(chainable: true)
+          .instance_variable_get('@query_proxy')
+          .instance_variable_get('@association_labels')
+      ).to eq(nil)
+
+      expect(
+        manager.favorite_subordinate(chainable: true)
+          .instance_variable_get('@query_proxy')
+          .instance_variable_get('@association_labels')
+      ).to eq(false)
+    end
+
+    # This is failing with a "NameError: undefined local variable or method `node'".
+    # As far as I can tell, this exact test works fine in console testing, making me think this is
+    # just a problem with the spec. My guess is it has to do with
+    # how the class is stubbed ?
+
+    # it 'allows overriding of default options' do
```

Comment:
This is the problem spec I was referring to

Reply:
If it's working in the console I'm fine with leaving this out for now.  There are definitely some hairy parts with stubbing in the gem (partially because of stubbing and partially because of various interdependencies within the modules, I think)

Reply:
@cheerfulstoic Should I leave the spec there but keep it commented out (as a note for people in the future)? Or should I just delete the commented out spec?

Reply:
Nvm, while addressing

> I wonder if this could be testing by calling to_cypher on the QueryProxy and doing a MATCH on the string perhaps

I was able to get the spec working

Reply:
For the curious: I think the issue was that I misinterpreted the ```NameError: undefined local variable or method `node` ```. The `has_one` method has an arg named `node`, and I thought the error was coming from there. In actuality, I think the error was coming from the node variable in the spec:
`expect(node.favorite_subordinte ...`

I think this was a copy-paste error from the `has_many` version of this spec.

Reply:
Awesome work, thanks!